src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/General/RoAssemblyName.cs

```diff
@@ -32,7 +32,7 @@ internal sealed class RoAssemblyName : IEquatable<RoAssemblyName>
 
         private static readonly Version s_Version0000 = new Version(0, 0, 0, 0);
 
-        public RoAssemblyName(string name, Version version, string cultureName, byte[] publicKeyToken, AssemblyNameFlags flags)
+        public RoAssemblyName(string? name, Version? version, string? cultureName, byte[]? publicKeyToken, AssemblyNameFlags flags)
         {
             // We forcefully normalize the representation so that Equality is dependable and fast.
             Debug.Assert(name != null);
```

Comment:
Isn't assert mean `name` shouldn't be nullable?  Which means `Name` non null and below GetHashCode() wouldn't need bang